### ISSUE TITLE
Dict[str, str] is not accepted as a subtype of a Mapping[str | istr, str] argument

### ISSUE THREAD

**Issue by ErwanDL** [microsoft/pyright#1246]
**Describe the bug**
I am creating a `web.Response` object in an aiohttp server project, which accepts a `headers` argument of type `Mapping[str | istr, str] | CIMultiDict[str] | CIMultiDictProxy[str] | None`. When trying to pass a plain string-to-string dictionary to this `headers` argument, the following error is raised by Pylance :

```
Argument of type "dict[str, str]" cannot be assigned to parameter "headers" of type "Mapping[str | istr, str] | CIMultiDict[str] | CIMultiDictProxy[str] | None" in function "__init__"
  Type "dict[str, str]" cannot be assigned to type "Mapping[str | istr, str] | CIMultiDict[str] | CIMultiDictProxy[str] | None"
    "dict[str, str]" is incompatible with "CIMultiDict[str]"
    "dict[str, str]" is incompatible with "CIMultiDictProxy[str]"
    Cannot assign to "None"
      TypeVar "_KT" is invariant
        Type "str" cannot be assigned to type "str | istr"Pylance (reportGeneralTypeIssues)
```
This looks like a bug, as `Dict[str, str]` should be accepted as a subtype of `Mapping[str | istr, str]`.

**To Reproduce**
Have [aiohttp](https://docs.aiohttp.org/en/stable/) installed in your environment, and paste the following code snippet : 
```python
from aiohttp import web

test_headers = {
    "Accept-Ranges": "bytes",
    "Content-Type": "audio/x-mpegurl",
}

res = web.Response(headers=test_headers)
```
Pylance will raise an error and add red squiggles at the end of the last line.

**Expected behavior**
There should be no type error here : `Dict[str, str]` should be accepted as a valid type for the `headers` parameter.

**VS Code extension or command-line**
Pylance extension (version: _Pylance language server 2020.12.0 (pyright ea581b3f)_)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.93, which I just published. It will also be included in the next release of Pylance.
